.github/workflows/metric.yml

```diff
@@ -1,17 +1,18 @@
 name: Metric and Commit Results
 
-permissions: {}
-
 on:
   workflow_dispatch: # Allows manual triggering of the workflow
-  push:
+  pull_request_target:
     branches:
       - main
     paths-ignore:
       - '**/*.md'
+    types:
+      - closed
 
 jobs:
   compile-time-and-binary-size:
+    if: github.event.pull_request.merged == true
```

Comment:
https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-your-pull_request_target-workflow-when-a-pull-request-merges